docker/package/tezos_baking_wizard.py

```diff
@@ -513,38 +560,79 @@ def bootstrap_node(self):
         print()
         print("The Tezos node bootstrapped successfully.")
 
+    # Check if an account with the 'baker' alias already exists, and ask the user
+    # if it can be overwritten.
+    def check_baker_account(self):
+        address = subprocess.run(
+            shlex.split("sudo -u tezos tezos-client show address baker --show-secret"),
+            capture_output=True,
+        )
+        if address.returncode == 0:
+            value_regex = b"(?:" + ledger_regex + b")|(?:" + secret_key_regex + b")"
+            value = re.search(value_regex, address.stdout).group(0)
+            address = re.search(address_regex, address.stdout).group(0)
+            print()
+            print("An account with the 'baker' alias already exists.")
+            print("Its current value is", value.decode("utf-8"))
+            print("With a corresponding address:", address.decode("utf-8"))
+
+            return yes_or_no("Should it be replaced with the new import? <y/N> ", "no")
+
     # Importing the baker key
     def import_baker_key(self):
-        if self.config["key_import_mode"] == "secret-key":
-            self.check_baker_account(
-                "sudo -u tezos tezos-client import secret key baker "
-                + self.config["secret_key"]
-            )
-        elif self.config["key_import_mode"] == "json":
-            self.check_baker_account(
-                "sudo -u tezos tezos-client activate account baker with "
-                + self.config["json_filepath"]
-            )
 
-        else:
-            print()
-            input("Please open the Tezos Baking app on your ledger, then hit Enter.")
-            self.check_baker_account(
-                "sudo -u tezos tezos-client import secret key baker "
-                + self.config["ledger_url"]
-            )
-            proc_call("sudo -u tezos tezos-client setup ledger to bake for baker")
+        replace_baker_key = self.check_baker_account()
+        valid_choice = False
+
+        while replace_baker_key and not valid_choice:
```

Comment:
Something seems off here, when there is no known baker account then `replace_baker_key` is `False` and so (regardless of `valid_choice`) this `while` loop never gets executed.

That, clearly, is a problem as what happens next is that `register_baker` is called and
`sudo -u tezos tezos-client register key baker as delegate` executed (which results in an error).